ocaml/bytecomp/bytegen.ml

```diff
@@ -1160,6 +1178,40 @@ and comp_expr_list stack_info env exprl sz cont = match exprl with
       comp_expr stack_info env exp sz
         (Kpush :: comp_expr_list stack_info env rem (sz+1) cont)
 
+(* Compile a list of reuse arguments [x; e1; ...; eN] and their resets
+   [None; Some _; ...] to a primitive operation. For each None, we push
+   a read on x on the stack and for each Some _, we push an ei on the stack.
+   As with regular arguments, we push in reversed order and
+   leave the first argument in the accumulator. *)
+
+and comp_reuse_args stack_info env argl resets getfield sz cont =
+  let rec zip_resets idx resets argl =
+    let next = idx + 1 in
+    match resets, argl with
+    | [], [] -> []
+    | Reuse_keep_old :: resets,        argl -> `ReadOld idx :: zip_resets next resets argl
+    | Reuse_set_to _ :: resets, arg :: argl -> `PushNew arg :: zip_resets next resets argl
+    | _ -> assert false (* There is exactly one arg for each Some. *)
+  in
+  match argl with
+  | [] -> assert false (* The first argument is the reused value. *)
+  | old :: argl ->
+    comp_reuse_expr_list stack_info env (List.rev (zip_resets 0 resets argl))
+      old getfield sz cont
+
+and comp_reuse_expr_list stack_info env exprl old getfield sz cont =
+  let comp_reuse_arg exp cont =
+    match exp with
+    | `ReadOld idx -> comp_expr stack_info env old sz (getfield idx :: cont)
+    | `PushNew exp -> comp_expr stack_info env exp sz cont
+  in
+  match exprl with
+  | [] -> cont
```

Comment:
Shouldn't this case be impossible?

Reply:
Because we never reuse blocks of size zero? I agree with this. However, this case is symmetric to the one in `comp_expr_list` which to my surprise can indeed happen, suggesting that the compiler creates `Pmakeblock`s of size 0.